packages/internals/renderer/src/adapter-store.js

```diff
@@ -0,0 +1,51 @@
+const {remove} = require('lodash');
+const {File} = require('@frctl/support');
+const {toArray} = require('@frctl/utils');
+const debug = require('debug')('fractal:renderer:adapters');
+const Adapter = require('./adapter');
+
+const _adapters = new WeakMap();
+
+class AdapterStore {
+
+  constructor(adapters = []) {
+    _adapters.set(this, []);
+    this.add(adapters);
+  }
+
+  getDefaultAdapter() {
+    return this.adapters[0];
+  }
+
+  getAdapterFor(file) {
+    if (!File.isFile(file)) {
+      throw new Error('Can only retrieve adapters for File objects [file-invalid]');
+    }
+    debug('Finding adapter for file %s: ', file.path);
+    for (const adapter of _adapters.get(this)) {
+      if (adapter.match(file)) {
+        return adapter;
+      }
+    }
+  }
+
+  add(items) {
+    const adapters = _adapters.get(this);
+    toArray(items).map(props => new Adapter(props)).forEach(adapter => {
```

Comment:
Tbh I'm not sure what we'd gain from that here... the error message thrown by the adapter is really the one that is useful (as it contains the validation errors) and seems a little overkill to catch and throw it again here.  I think I will prob leave like this for now and we can always revisit if it's confusing later.